kmk/extensions/oled.py

```diff
@@ -0,0 +1,291 @@
+        flip_right: bool = False,
+        device_address=0x3C,
+        brightness=0.8,
+        brightness_step=0.1,
+        dim_time=None,
+        off_time=None,
+        powersave_dim_time=10,
+        powersave_off_time=60,
+    ):
+        displayio.release_displays()
+        self._flip = flip
+        self._flip_left = flip_left
+        self._flip_right = flip_right
+        self._views = views.data
+        self._width = width
+        self._height = height
+        self._prevLayers = 0
+        self._device_address = device_address
+        self._brightness = brightness
+        self._brightness_step = brightness_step
+        self._timer_start = ticks_ms()
+        self._dark = False
+        self._go_dark = False
+        self._powersave = False
+        self._dim_time_ms = dim_time * 1000 if dim_time else None
+        self._off_time_ms = off_time * 1000 if off_time else None
+        self._powersave_dim_time_ms = (
+            powersave_dim_time * 1000 if powersave_dim_time else None
+        )
+        self._powersave_off_time_ms = (
+            powersave_off_time * 1000 if powersave_off_time else None
+        )
+        self._split = split
+
+        make_key(
+            names=('OLED_BRI',), on_press=self._oled_bri, on_release=handler_passthrough
+        )
+        make_key(
+            names=('OLED_BRD',), on_press=self._oled_brd, on_release=handler_passthrough
+        )
+
+    def render_oled(self, layer, *args, **kwargs):
+        splash = displayio.Group()
+
+        for view in self._views:
+            if self._dark is False:
+                if view[4] == layer or view[4] is None:
+                    if view[5] == OledEntryType.TXT:
+                        splash.append(
+                            label.Label(
+                                terminalio.FONT,
+                                text=view[0],
+                                color=0xFFFFFF if not view[10] else 0x000000,
+                                background_color=0x000000 if not view[10] else 0xFFFFFF,
+                                anchor_point=(view[6], view[7]),
+                                anchored_position=(view[1], view[2]),
+                                label_direction=view[8],
+                                line_spacing=view[9],
+                                padding_left=1,
+                            )
+                        )
+                    elif view[5] == OledEntryType.IMG:
+                        splash.append(
+                            displayio.TileGrid(
+                                view[0],
+                                pixel_shader=view[0].pixel_shader,
+                                x=view[1],
+                                y=view[2],
+                            )
+                        )
+        gc.collect()
+        self._display.show(splash)
+
+    def updateOLED(self, sandbox):
+        self.render_oled(sandbox.active_layers[0])
+        gc.collect()
```

Comment:
There may be some memory beeing freed here, but still. Verify that the code won't run without.